### PR TITLE
Fix library call test

### PR DESCRIPTION
**Pull request by MaksymilianDemitraszek**
Closes: #697 

## Introduced changes

- library calls now work in tests

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/cheatnet/src/constants.rs [hunk 1]
```diff
@@ -187,19 +192,25 @@ pub fn build_testing_state(predeployed_contracts: &Utf8PathBuf) -> DictStateRead
     let block_context = build_block_context();
     let test_account_class_hash = ClassHash(stark_felt!(TEST_ACCOUNT_CONTRACT_CLASS_HASH));
     let test_erc20_class_hash = ClassHash(stark_felt!(TEST_ERC20_CONTRACT_CLASS_HASH));
+    let test_contract_class_hash = ClassHash(stark_felt!(TEST_CONTRACT_CLASS_HASH));
 
     let class_hash_to_class = HashMap::from([
-        (test_account_class_hash, ContractClass::V1(account_class)),
+        (test_account_class_hash, ContractClass::V1(account_class.clone())),
+        // This is dummy put here only to satisfy blockifier 
+        // this class is not used and the test contract cannot be called
```
Comment (Arcticae) [inline crates/cheatnet/src/constants.rs:200]: Can we add a test which checks this?

**Review by Arcticae — CHANGES_REQUESTED** [review]

File: crates/cheatnet/src/constants.rs [hunk 2]
```diff
@@ -45,6 +45,11 @@ pub const TEST_FAULTY_ACCOUNT_CONTRACT_CLASS_HASH: &str = "0x113";
 pub const SECURITY_TEST_CLASS_HASH: &str = "0x114";
 pub const TEST_ERC20_CONTRACT_CLASS_HASH: &str = "0x1010";
 
+pub const TEST_CONTRACT_CLASS_HASH: &str = "0x117";
+// snforge_std/src/cheatcodes.cairo::TEST
```
Comment (piotmag769) [inline crates/cheatnet/src/constants.rs:49]:
```suggestion
// snforge_std/src/cheatcodes.cairo::test_contract
```

**Review by piotmag769 — APPROVED** [review]
LGTM, please add the test @Arcticae mentioned though

File: crates/cheatnet/src/constants.rs (hunk 2 above)
Comment (MaksymilianDemitraszek) [reply crates/cheatnet/src/constants.rs:49]:
```suggestion
// snforge_std/src/cheatcodes.cairo::test_address
```

File: crates/cheatnet/src/constants.rs (hunk 1 above)
Comment (MaksymilianDemitraszek) [reply crates/cheatnet/src/constants.rs:200]: If you mean one checking if it cannot be called, such one already exists https://github.com/foundry-rs/starknet-foundry/blob/581ac740b5bf360f9df4b3fa63a83ece9ee564c6/crates/forge/tests/integration/test_state.rs#L336

**Review by piotmag769 — APPROVED** [review]

**Review by Arcticae — APPROVED** [review]

File: crates/cheatnet/src/constants.rs (hunk 1 above)
Comment (MaksymilianDemitraszek) [reply crates/cheatnet/src/constants.rs:202]: I updated the contract to be fully dummy as actually this changes made this test fail

**Pull request merged**